### PR TITLE
adding condition as requsted by AWS

### PR DESCRIPTION
**Pull request by rafael-azevedo**
This PR adds requested condition to image registry policy that ensure that the requesting account is the account that owns the resource.

### CODE REVIEW

File: resources/sts/4.14/hypershift/openshift_hcp_image_registry_operator_permission_policy.json [hunk 1]
```diff
@@ -8,7 +8,12 @@
         "s3:ListBucket",
         "s3:ListBucketMultipartUploads"
       ],
-      "Resource": "*"
+      "Resource": "*",
+      "Condition": {
+        "StringEquals": {
+            "aws:ResourceAccount": "${aws:PrincipalAccount}"
+        }
+    }
```
Comment (dustman9000) [inline resources/sts/4.14/hypershift/openshift_hcp_image_registry_operator_permission_policy.json:16]: Fix indentation

**Review by mjlshen — COMMENTED** [review]
/lgtm

File: resources/sts/4.14/hypershift/openshift_hcp_image_registry_operator_permission_policy.json [hunk 2]
```diff
@@ -28,7 +33,12 @@
       ],
       "Resource": [
         "arn:aws:s3:::*-image-registry-${aws:RequestedRegion}*"
-      ]
+      ],
+      "Condition": {
+        "StringEquals": {
+            "aws:ResourceAccount": "${aws:PrincipalAccount}"
+       }
```
Comment (dustman9000) [inline resources/sts/4.14/hypershift/openshift_hcp_image_registry_operator_permission_policy.json:40]:
```suggestion
        }
```

**Pull request merged**